UnityProject/Assets/Scripts/Objects/Research/BlastYieldDetector.cs

```diff
@@ -110,7 +110,7 @@ private void DetectBlast(Vector3Int pos, BlastData blastData)
 				{
 					if (reaction.IsReactionValid(blastData.ReagentMix))
 					{
-						yield += (float)(-463 + 205 * Mathf.Log(reaction.GetReactionAmount(blastData.ReagentMix) + 75 * MathF.PI)); //Formula taken from strength calculation in ChemExplosion.cs, This will flag codacy cause magic numbers, but idk what these numbers mean or what they relate to so I dont wanna create random constants.
+						yield += 32 * Mathf.Pow(reaction.GetReactionAmount(blastData.ReagentMix), 0.6f); //Taken from ChemExplosion.cs
```

Comment:
why not turn this into a function and return it from chemexplosion.cs?

Reply:
assemblies and cyclic assmebly references means I cant reference chemexplosion.cs

Reply:
the chemistry sub assemblies are arranged in a really wierd way and it annoys me alot

Reply:
Maybe that happens because Chen's reference the shared assembly. For it to work you would need to mo e your thing into its own assembly, outside the main one or move that function to the shared assembly, outside Chems.

Reply:
its not the shred assmebly. Its because Chemistry.Effects isnt actually an assembly def. Its an entirely seperate assembly that references chemistry and assets. So If I try reference Chemistry.Effects from Assets it causes cyclic. Its the same with all the chemistry assemblies, whoever set them up set them up strange. It could probably be reworked, but that's a good few hours work restructuring chemistry to save one line of code. Which I personally would rather avoid

Reply:
Assets is the shared assembly. Everything exists there.

Reply:
Dont we have assets and shared assemblies

Reply:
Ah we have an assembly called shared, I see. What I meant is that it isn't chems what is wrongly set. It is everything else that shouldn't be in assets assembly. 

That said, you have the option of moving the function to the so called shared assembly and it shouldn't be so painful then.

Reply:
Found out we have a ChemistryUtils class so Ive put the function in there